### PR TITLE
Change navigation on MyAccount Chiturca/issue103

### LINKED ISSUE #103: Change navigation on MyAccount page 🐞
---
about: signed in user main page blog navigation 
---

## Description

When signed in user clicked on the blog, it navigates to WriteBlog page directly instead of Blog.jsx.

## Steps to Reproduce

1. Change navigation to Blog.jsx if user is signed in
2. Change navigation to WriteBlog.jsx if user has no blogs 

**Expected behavior:** When user click on their own blogs, they should see the details

**Actual behavior:** User's own blogs navigates user to write a new blog

### PR DESCRIPTION
**Pull request by chiturca**
# Description

The cards when user have blog, has been changed to BlogCard component and MyAccountCard has been changed in order to be a placeholder for user has no blogs and they are connected with ternary.

## Related Issue

Fixes #103

## Screenshots
user has blogs
user has no blogs

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally
- [ ] Added tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [x] I've run linter, prettier, and removed all `console.log`
- [x] I've reverted all the files back that are not related to the issue (App.js, index.js, ...)

### CODE REVIEW

File: src/pages/UserProfile/MyAccount.jsx [hunk 1]
```diff
@@ -42,30 +53,29 @@ function MyAccount() {
...
-                  .filter((blog) => blog.data.userID === user.id)
-                  .map((blog, i) => {
-                    return <MyAccountCard key={i} data={blog.data} />;
-                  })}
+                  ? blogs
+                      .filter((blog) => blog.data.author.authorId === user.id)
+                      .map((blog, i) => {
+                        return <BlogCard key={i} blog={blog} />;
+                      })
+                  : emptyCard.map((holder) => (
```
Comment (ahmadmhd357) [inline src/pages/UserProfile/MyAccount.jsx:76]: I couldn't understand this part

Comment (chiturca) [reply src/pages/UserProfile/MyAccount.jsx:76]: To sum up,
*If user has blogs, show their blog data (which connects to BlogCard.jsx) 
*If thats not the case (i mean if user has no blog) show another, empty card to user so she/he can click on that card and start to write in WriteBlog.  And, they could know that their own blogs will be shown in that part of the MyAccount component, you know? 

I mean a new user will not know about our site structure so I tried to do something like this.
So this way I thought that MyAccount page would have been seen fully from the user's side

**Review by chiturca — COMMENTED** [review]
This looks better right? @ahmadmhd357

**Comment by ahmadmhd357** [comment]
> This looks better right? @ahmadmhd357

yes and as another approach, you could just delete the mapping and put the card itself as component

**Review by ahmadmhd357 — APPROVED** [review]

**Review by Omer-Amr — APPROVED** [review]
looks good thank you Miray!

**Pull request merged**